R/table.R

```diff
@@ -130,12 +139,19 @@ html_table.xml_node <- function(x,
     col_names <- paste0("X", seq_len(ncol(out)))
   }
 
-  # Convert matrix to list to data frame
-  df <- lapply(seq_len(ncol(out)), function(i) {
-    utils::type.convert(out[, i], as.is = TRUE, dec = dec, na.strings = na.strings)
-  })
-  names(df) <- col_names
-  tibble::as_tibble(df, .name_repair = "minimal")
+  if (isTRUE(convert)) {
+    # With type conversion, convert matrix to list to data frame
+    df <- lapply(seq_len(ncol(out)), function(i) {
+      utils::type.convert(out[, i], as.is = TRUE, dec = dec, na.strings = na.strings)
+    })
+
+    names(df) <- col_names
```

Comment:
Could you move these two lines outside of the if block? (By overwriting `out` above). Then I think you wouldn't need the else statement.

Reply:
Do you mean something like this? If so, I too considered this but `names()` cannot be used on a `matrix` and `colnames()` cannot be used on a `list`...

``` r
if (isTRUE(convert)) {
  # With type conversion, convert matrix to list to data frame
  out <- lapply(seq_len(ncol(out)), function(i) {
    utils::type.convert(out[, i], as.is = TRUE, dec = dec, na.strings = na.strings)
  })
}

names(out) <- col_names
tibble::as_tibble(out, .name_repair = "minimal")
```

Reply:
Oooh, good point. 

You could do `out[] <- lapply(...)` so that `out` stays as a data frame?

Reply:
Oh but it's a matrix so that won't work.

Maybe it would be better to convert it to a tibble, name it, and _then_ convert the columns?

Reply:
Thanks, I didn't come up with `out[] <- lapply(...)`! Now the implementation looks cleaner :)